### PR TITLE
vnoc-8591 create test bench

### PR DESCRIPTION
**Pull request by ducanhpham0312**
### Description
- A Docker environment with Zeebe (with embedded gateway), zeebe-kafka-exporter, Kafka, Postgres, and Zeebe Simple Monitor.
- Some process files in BPMN format ready to be deployed to Zeebe
- Some pre-defined job workers to be called from BPMN files.
- A detailed README.

### Testing
- Start the docker environment: `./start.sh`.
- Start job worker: Run job workers in IntelliJ
- Process and process instance can be deployed and created using `zbctl`. Instruction on using it is available in README.
- Job worker and message publisher can also be handled using `zbctl`. More details in README.
- Status of the instance can be observed in Zeebe Simple Monitor.
- The README should be detailed and easy to follow, even if one has little knowledge in development. University students will be the targeted audience of this repo.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    
+  kafka:
+    image: bitnami/kafka:3.1.0
+    ports:
+      - "9092:9092"
+    environment:
+      - KAFKA_BROKER_ID=0
+      - ALLOW_PLAINTEXT_LISTENER=yes
+      - "KAFKA_HEAP_OPTS=-Xmx500m -Xms500m"
+      - KAFKA_CFG_ZOOKEEPER_CONNECT=zookeeper:32181
```
Comment (Ninjaluo) [inline docker-compose.yml]: Changed this to port 2181 as specified on line 9 and added dependency to zookeeper to get it work on intel machine
```
    depends_on:
      - zookeeper
```

File: docker-compose-arm.yml [hunk 2]
```diff
@@ -68,8 +68,8 @@ services:
       - "26500:26500"
       - "9600:9600"
     volumes:
-      - ./config/zeebe/application.yaml:/usr/local/zeebe/config/application.yaml
-      - ./exporter/zeebe-kafka-exporter-3.1.1-jar-with-dependencies.jar:/usr/local/zeebe/lib/zeebe-kafka-exporter.jar
+      - ./config/zeebe/application-arm.yaml:/usr/local/zeebe/config/application.yaml
+      - ./exporters/zeebe-kafka-exporter-3.1.1-jar-with-dependencies.jar:/usr/local/zeebe/lib/zeebe-kafka-exporter.jar
```
Comment (Ninjaluo) [inline docker-compose-arm.yml:72]: Now the setup works for intel 👍 Remember to update the readme as well

File: docker-compose-arm.yml [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+      - KAFKA_ZOOKEEPER_CONNECT=ignored
+    volumes:
+      - ./topics:/tmp/topics.txt
+      - ./create-topics.sh:/tmp/create-topics.sh
+    entrypoint: [ '/bin/bash', '-c' ]
+    command: sh /tmp/create-topics.sh
+    networks:
+      - devenv
+  
+  # get-kafka-exporter:
```
Comment (jaakkoraut) [inline docker-compose-arm.yml]: This is commented out? Could you remove it or fix it.

File: docker-compose-arm.yml [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    volumes:
+      - ./config/zeebe/application-arm.yaml:/usr/local/zeebe/config/application.yaml
+      - ./exporters/zeebe-kafka-exporter-3.1.1-jar-with-dependencies.jar:/usr/local/zeebe/lib/zeebe-kafka-exporter.jar  
+    depends_on:
+      - kafka
+    networks:
+      - devenv
+
+  zeebe-simple-monitor:
+    image: ghcr.io/elisapolystar/zeebe-simple-monitor:latest
```
Comment (jaakkoraut) [inline docker-compose-arm.yml:79]: 👍

File: start.sh [hunk 5]
```diff
@@ -0,0 +1,29 @@
+arch=$(uname -m)
```
Comment (jaakkoraut) [inline start.sh:1]: Would it make sense to provide a startup script (.bat) for Windows users?

File: zeebe-job-worker/src/main/kotlin/job/moneyloan/SendAcceptanceLetter.kt [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+import io.camunda.zeebe.client.ZeebeClient
+import io.camunda.zeebe.client.ZeebeClientBuilder
+import io.camunda.zeebe.client.api.response.ActivatedJob
+import io.camunda.zeebe.client.api.worker.JobClient
+import io.camunda.zeebe.client.api.worker.JobHandler
+import java.time.Duration
+import java.util.Scanner
+
+/**
+ * When connecting to a cluster in Camunda Cloud, this application assumes that the following
```
Comment (jaakkoraut) [inline zeebe-job-worker/src/main/kotlin/job/moneyloan/SendAcceptanceLetter.kt]: to Zeebe cluster (same issue in various files)

File: zeebe-job-worker/src/main/kotlin/job/multiinstance/InitiatePayment.kt [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+ *  * ZEEBE_CLIENT_ID
+ *  * ZEEBE_CLIENT_SECRET
+ *  * ZEEBE_AUTHORIZATION_SERVER_URL
+ *
+ * When `ZEEBE_ADDRESS` is not set, it connects to a broker running on localhost with
+ * default ports
+ */
+object InitiatePayment {
+    @JvmStatic
+    fun main(args: Array<String>) {
```
Comment (jaakkoraut) [inline zeebe-job-worker/src/main/kotlin/job/multiinstance/InitiatePayment.kt:19]: All these have separate main functions, wouldn't it be easier to start the service in one main and then expose all of these

**Review by jaakkoraut — CHANGES_REQUESTED** [review]
Left some comments and here's one more:
- please remove all internal references from PR (ticket, etc)
- make sure repository is compliant with the derivate works included, I noticed at least Zeebe Community license 1.1 and Apache license referred
- license the repository as well so that it's compliant with all the used content

With these changes, I think this will be excellent!

File: start.sh (hunk 5 above)
Comment (ducanhpham0312) [reply start.sh:1]: The updated README has additional instructions for Windows to install WSL (Windows Subsystem for Linux which basically an additional layer that allow users to run Linux in Windows, providing tools and application etc. WSL can run shell script smoothly without any issues.

**Pull request closed without merging**